src/servant/tt_001/F84.c

```diff
@@ -14,7 +14,34 @@ INCLUDE_ASM("servant/tt_001/nonmatchings/F84", func_us_80171284);
 
 void func_us_80171560(void) {}
 
-INCLUDE_ASM("servant/tt_001/nonmatchings/F84", func_us_80171568);
+Entity* func_us_80171568(Entity* self, s32 entityIdFlag) {
+    Entity* entity;
+    s32 i;
+
+    for (i = 0; i < 3; i++) {
+        entity = &D_80073784[i];
+        if (entity->entityId == 0) {
+            break;
+        }
+    }
+
+    if (entity->entityId == 0) {
+        DestroyEntity(entity);
+
+        if (entityIdFlag == 0) {
+            entity->entityId = 0xDA;
+        } else {
+            entity->entityId = 0xDB;
+        }
+        entity->zPriority = self->zPriority;
+        entity->facingLeft = self->facingLeft;
+        entity->flags = FLAG_KEEP_ALIVE_OFFCAMERA;
+        entity->posX.val = self->posX.val;
+        entity->posY.val = self->posY.val;
+        entity->ext.factory.parent = self;
+        return entity;
+    }
+}
```

Comment:
Seems strange to me that there is a fall-through case here which doesn't return a value, maybe I have something wrong?

Reply:
This gets both versions matching with some #ifdefs, maybe it can be cleaned up more?

https://decomp.me/scratch/rbZPM
https://decomp.me/scratch/K9ZjY

Reply:
I bet these same differences exist in HD vs. US

Reply:
Good call, thanks

Reply:
Also to answer your question there's other spots in the codebase where an undefined return value is apparently required to match, just seems like it was that way originally